### PR TITLE
Bulk JSON typo fixes

### PR DESCRIPTION
**Pull request by BrettDong**
#### Summary
None

#### Describe the solution
Those are misspellings found by comparing to an English corpus dataset and search for unrecognised tokens with Levenshtein distance less than or equal to 2 to a known token in the corpus.

### CODE REVIEW

File: data/json/mapgen/mapgen-test.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   {
     "id": "mapgen-test",
     "type": "overmap_terrain",
-    "name": "noname",
+    "name": "no name",
```
Comment (Night-Pryanik) [inline data/json/mapgen/mapgen-test.json:20]: I guess this is unneeded.

Comment (BrettDong) [reply data/json/mapgen/mapgen-test.json:20]: This is to reduce the unwanted noise in CDDA corpus & word frequencies dataset I'm working on next.

Comment (Night-Pryanik) [reply data/json/mapgen/mapgen-test.json:20]: Ok, got it.

Comment (BrettDong) [reply data/json/mapgen/mapgen-test.json:20]: By the way I just realised this should be moved to TEST DATA and should not be translated.

**Review by Maleclypse — COMMENTED** [review]
AFS changes are all good I think.

**Pull request merged**